qna/src/app/parser/parser.py

```diff
@@ -51,6 +58,25 @@ def getPostsFromThreads(threads):
     :param threads: List of question/answer strings
     :return None:
     '''
+
+    module_url = "https://tfhub.dev/google/universal-sentence-encoder/4"
+    model = hub.load(module_url)
+    print("Finished Loading Universal Encoder")
+    embeddings_questions = model(
+        [email.message_from_string(i)['Subject'] for i in threads])
+    print("Finished Universal Encoder Embeddings")
+
+    model2 = T5ForConditionalGeneration.from_pretrained("t5-small")
+    tokenizer = T5Tokenizer.from_pretrained("t5-small")
+    print("Finished Loading T5 model")
+
+    start = time.time()
+    embeddings_text = model([email.message_from_string(i)['Subject'] + get_summarisation(
+        email.message_from_string(i)._payload, tokenizer, model2) for i in threads])
+    end = time.time() - start
+    print(end)
+    print("Finished t5 Embeddings")
+
```

Comment:
Testing this out it seems that it is calculating the embeddings and summarisations for every post (both questions and answers). 

With the embeddings this doesn't seem too bad since it is very quick but each summarisation (on my machine) takes ~1-1.5s. Might be worth reworking this so that we only calculate summaries for the questions?